### PR TITLE
feat(raw): raw reader - exposing max_raw_memory_mb

### PR DESCRIPTION
**Pull request by ankmachine**
## Description

PR is to address https://github.com/AcademySoftwareFoundation/OpenImageIO/issues/3881 feature request. The reported wanted to expose `imgdata.rawparams.max_raw_memory_mb` as `raw:max_raw_memory_mb`

## Tests

I was able to compile my code successfully, no new test were added for this.

## Checklist:

- [x] I have read the [contribution guidelines](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/main/CONTRIBUTING.md).
- [x] I have updated the documentation, if applicable. (Check if there is no
  need to update the documentation, for example if this is a bug fix that
  doesn't change the API.)
- [ ] I have ensured that the change is tested somewhere in the testsuite
  (adding new test cases if necessary).
- [ ] If I added or modified a C++ API call, I have also amended the
  corresponding Python bindings (and if altering ImageBufAlgo functions, also
  exposed the new functionality as oiiotool options).
- [ ] My code follows the prevailing code style of this project. If I haven't
  already run clang-format before submitting, I definitely will look at the CI
  test that runs clang-format and fix anything that it highlights as being
  nonconforming.

### CODE REVIEW

File: src/raw.imageio/rawinput.cpp [hunk 1]
```diff
@@ -453,6 +453,14 @@ RawInput::open_raw(bool unpack, const std::string& name,
...
     m_processor->imgdata.params.output_bps = 16;
 
+    // exposing max_raw_memory_mb setting
+    static_assert(sizeof(m_processor->imgdata.rawparams.max_raw_memory_mb)
+                      == sizeof(int),
+                  "Sizes must match");
+    config.getattribute("raw:max_raw_memory_mb", TypeDesc::INT,
+                        &m_processor->imgdata.rawparams.max_raw_memory_mb);
+
+
```
Comment (lgritz) [inline src/raw.imageio/rawinput.cpp:463]: Might it be simpler to express this more like

    m_processor->imgdata.rawparams.max_raw_memory_mb = 
        config.get_int_attribute("raw:max_raw_memory_mb", <default value>);

?? Then you wouldn't need the static assert, and various type conversions would happen automatically if the types ever change.

Also, would you mind please documenting the existence and meaning of this attribute in the raw section of builtinplugins.rst ? Thanks.

**Review by lgritz — CHANGES_REQUESTED** [review]
This is great, I just made a couple suggestions.

File: src/raw.imageio/rawinput.cpp (hunk 1 above)
Comment (lgritz) [reply src/raw.imageio/rawinput.cpp:463]: What's a good default for this? The other reason I like my suggestion above is that it's probably a good idea to by default impose some reasonable limit, even if this parameter is not passed from the user. We shouldn't allow big files to consume unlimited memory in libraw unless specifically asked to do so.

Comment (JenusL) [reply src/raw.imageio/rawinput.cpp:463]: Default in libraw is 2048 (2GB)
https://github.com/LibRaw/LibRaw/blob/9bcb8a1d9593ba67e4eb67fed716efc5e1353d5c/libraw/libraw_const.h#L24
and
https://github.com/LibRaw/LibRaw/blob/9bcb8a1d9593ba67e4eb67fed716efc5e1353d5c/doc/API-datastruct.html#L556-L558
Libraw itself doesn't have a upper limit.

Comment (lgritz) [reply src/raw.imageio/rawinput.cpp:463]: Great, let's just express that default as I suggested above, then you don't need the static_assert or the needing pointers.

**Comment by ankmachine** [comment]
> This is great, I just made a couple suggestions.

@lgritz I have addressed the feedback and added the documentation. Let me know if its good :) .

**Review by lgritz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**